### PR TITLE
Remove generated alerts

### PR DESCRIPTION
**Pull request by Fokko**
While working on upgrading the Avro dependency, I've noticed that there are a lot of warning, that can easily be resolved:

- Unused variables/methods
- Unnecessary parentheses
- Removed unchecked warning

Makes the output a bit less verbose :)

### CODE REVIEW

File: flink/src/main/java/org/apache/iceberg/flink/RowDataTaskWriterFactory.java [hunk 1]
```diff
@@ -127,7 +126,7 @@ private FlinkFileAppenderFactory(Schema schema, RowType flinkSchema, Map<String,
     @Override
     public FileAppender<RowData> newAppender(OutputFile outputFile, FileFormat format) {
       // TODO MetricsConfig will be used for building parquet RowData writer.
-      MetricsConfig metricsConfig = MetricsConfig.fromProperties(props);
+      // MetricsConfig metricsConfig = MetricsConfig.fromProperties(props);
```
Comment (RussellSpitzer) [inline flink/src/main/java/org/apache/iceberg/flink/RowDataTaskWriterFactory.java:129]: My only question here is why not just delete the line and todo if we aren't using them and just have a github issue instead?

**Comment by rdblue** [comment]
Looks like this is failing Java tests.

File: flink/src/main/java/org/apache/iceberg/flink/RowDataTaskWriterFactory.java (hunk 1 above)
Comment (kbendick) [reply flink/src/main/java/org/apache/iceberg/flink/RowDataTaskWriterFactory.java:129]: I agree here. A github issue is more likely to be picked up and worked on over a TODO in the codebase.... though I'm not opposed to keeping the TODO so people don't accidentally overly refactor away from the possibility (if that's a potential concern).

File: spark3/src/main/java/org/apache/iceberg/spark/source/SparkBatchScan.java [hunk 2]
```diff
@@ -373,12 +372,5 @@ private Schema expectedSchema() {
       }
       return expectedSchema;
     }
-
-    private NameMapping nameMapping() {
-      if (nameMapping == null) {
-        this.nameMapping = NameMappingParser.fromJson(nameMappingString);
-      }
-      return nameMapping;
-    }
```
Comment (kbendick) [inline spark3/src/main/java/org/apache/iceberg/spark/source/SparkBatchScan.java:382]: I think there might be a PR open related to this. However, I don't see this particular function used: https://github.com/apache/iceberg/pull/1399/files

File: flink/src/main/java/org/apache/iceberg/flink/RowDataTaskWriterFactory.java (hunk 1 above)
Comment (Fokko) [reply flink/src/main/java/org/apache/iceberg/flink/RowDataTaskWriterFactory.java:129]: Good point, and I fully agree. I've created a Github issue: https://github.com/apache/iceberg/issues/1402

File: spark3/src/main/java/org/apache/iceberg/spark/source/SparkBatchScan.java (hunk 2 above)
Comment (Fokko) [reply spark3/src/main/java/org/apache/iceberg/spark/source/SparkBatchScan.java:382]: I think we can remove this for now.

**Pull request closed without merging**